src/App.tsx

```diff
@@ -60,11 +61,17 @@ const App: React.FC = () => {
     );
 };
 
-type SquareAddress = number;
+export type SquareAddress = number;
 
-type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
+export type SquareValue = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 ;
 
-type NumberMode = "normal" | "corner";
+export type NumberMode = "normal" | "corner";
+
+export type Values = Map<SquareAddress, SquareValue>;
+
+export type Groupings = SquareAddress[][];
+
+export type Contradictions = Set<SquareAddress>;
```

Comment:
All exported types must have jsdocs.

Reply:
Secondly, when you have shared types across multiple files, it's useful to extract them into a single `types.ts` file.